packages/chopsticks/src/context.ts

```diff
@@ -34,5 +35,15 @@ export const setupContext = async (argv: Config, overrideParent = false) => {
   await overrideStorage(chain, argv['import-storage'], at)
   await overrideWasm(chain, argv['wasm-override'], at)
 
+  // load blocks from db
+  if (chain.db && argv.resume) {
+    const blocks = await chain.db.getRepository(BlockEntity).find({ where: {}, order: { number: 'asc' } })
```

Comment:
this is not enough. what happens if there're old blocks not related to our fork.
The easy way I can think is to resume if first db block is head+1 otherwise they must be considered invalid, also starting without resume should clear blocks from db and have a fresh start

Reply:
> what happens if there're old blocks not related to our fork

we can let user to make sure the db file they are using is the correct one, do you think this is possible?

> resume if first db block is head+1

for `head`, do you mean the `block` in cli option? if it defaults to the latest block, then we won't load blocks in db, is this what we want?

> starting without resume should clear blocks from db and have a fresh start

Yes, this make sense

Reply:
> we can let user to make sure the db file they are using is the correct one, do you think this is possible?
> 
> >

No we can't because if I start once at block 100 and build 2 blocks then second time start at block 105 build 2 more blocks then I will have 101, 102, 106, 107 in database which are from same chain but not related

Reply:
> for `head`, do you mean the `block` in cli option? if it defaults to the latest block, then we won't load blocks in db, is this what we want?

after chain is setup, `chain.head`